### PR TITLE
Introduced MediaGallerySynchronization modules

### LINKED ISSUE #1147: Rename AssetIndexer
The operation that is being performed is rather synchronization that reindex, thus it should be renamed.
https://github.com/magento/adobe-stock-integration/blob/fbc4a5ce412a4e224c73f0879d97b0fdf76656d3/MediaGalleryUi/Model/AssetIndexer.php#L22

### Acceptance Criteria:
- `AssetIndexer` is renamed to `AssetSynchronization`
- All related logic is renamed to reflect that (CLI command, etc)
- Synchronization logic is moved out form `MediaGalleryUi` module

### PR DESCRIPTION
**Pull request by coderimus**
Introduced new modules `Magento_MediaGallerySynchronization` and `Magento_MediaGallerySynchronizationApi`, extracted sync logic to those modules

### Fixed Issues (if relevant)
magento/adobe-stock-integration#1147: Rename AssetIndexer
magento/adobe-stock-integration#1213: The "Source - Uploaded locally" filter returns zero images
magento/adobe-stock-integration#1148: Create new configuration

### CODE REVIEW

**Comment by coderimus** [comment]
@sivaschenko @lenaorobei 
As I can see this PR https://github.com/magento/adobe-stock-integration/pull/1158 keeps the `ExcludedDirectories`  under the `MediaGalleryUi` If this is a decision, this current PR depends on the mentioned one.

File: AdobeStockImage/Model/AssetIndexer.php [hunk 1]
```diff
@@ -14,14 +14,13 @@
...
 use Magento\Framework\Filesystem\DriverInterface;
 use Magento\MediaGalleryApi\Model\Asset\Command\GetByPathInterface;
-use Magento\MediaGalleryUi\Model\Filesystem\IndexerInterface;
+use Magento\MediaGallerySynchronizationApi\Api\AssetImagesSynchronizationInterface;
 
 /**
  * Check is image licensed and save it to media gallery asset grid
  */
-class AssetIndexer implements IndexerInterface
+class AssetIndexer implements AssetImagesSynchronizationInterface
```
Comment (sivaschenko) [inline AdobeStockImage/Model/AssetIndexer.php]: I think this class can be removed as asset's grid was eliminated

File: MediaGallerySynchronization/Model/AssetImagesSynchronization.php [hunk 2]
```diff
@@ -5,21 +5,21 @@
...
 use Magento\MediaGalleryApi\Model\Asset\Command\SaveInterface;
-use Magento\MediaGalleryUi\Model\Filesystem\IndexerInterface;
+use Magento\MediaGallerySynchronizationApi\Api\AssetImagesSynchronizationInterface;
 use Psr\Log\LoggerInterface;
 
 /**
  * Create MediaGallery asset and save it to database based on file information
  */
-class AssetIndexer implements IndexerInterface
+class AssetImagesSynchronization implements AssetImagesSynchronizationInterface
```
Comment (sivaschenko) [inline MediaGallerySynchronization/Model/AssetImagesSynchronization.php]: Let's simplify the name of the class and interface
```suggestion
class ImagesSynchronization implements ImagesSynchronizationInterface
```

File: MediaGallerySynchronization/Model/AssetImagesSynchronization.php [hunk 3]
```diff
@@ -75,10 +75,7 @@ public function __construct(
     }
 
     /**
-     * Create MediaGallery asset and save it to database based on file information
-     *
-     * @param \SplFileInfo $item
-     * @throws \Exception
+     * @inheritDoc
      */
     public function execute(\SplFileInfo $item): void
```
Comment (sivaschenko) [inline MediaGallerySynchronization/Model/AssetImagesSynchronization.php]: Please change the method to work with multiple `$items` (\SplFileInfo[])
```suggestion
    public function execute(array $items): void
```

File: MediaGalleryUi/composer.json [hunk 4]
```diff
@@ -9,6 +9,7 @@
         "magento/module-store": "*",
         "magento/module-media-gallery-ui-api": "*",
         "magento/module-media-gallery-api": "*",
+        "magento/module-media-gallery-synchronization-api": "*",
```
Comment (sivaschenko) [inline MediaGalleryUi/composer.json:12]: I don't think MediaGalleryUi should depend on synchronization, please reverse the dependency

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @coderimus ! Please take a look at my review comments

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
Functional Tests B2B failure is not related to changes and randomly failing on other PRs as well: `Magento\FunctionalTestingFramework.functional\MSI_Multi_Mode.AdminCreateShipmentWithChangedSimpleProductIntoConfigurableDefaultStockTest`

MHI failure is related to the protected method that was extended from Symfony command

**Pull request merged**